### PR TITLE
Select Text

### LINKED ISSUE #256: Share as image/select text
There were two great sharing features in Apollo:

Select text - when you haptic touched on a comment, there was a select text button which was super handy! If a comment wasn’t in English, you could just select the text and use the built in translate feature. Or if you didn’t know what a word meant, you could look it up! Or you could just copy the text. It was great!

Share as image - Another handy sharing feature that I used to use alllll the time was share as image. You could Haptic Touch one a comment or post and tap share as image and it would render it out into a nice well formatted image to share with anyone!

### PR DESCRIPTION
**Pull request by Sjmarf**
Closes #256

This adds the [SwiftUIIntrospect](https://github.com/siteline/swiftui-introspect) library as a dependency. I think this is worth having - I also plan to use it in future for detecting the cursor location within a TextField for inline image uploading.

<hr>

### CODE REVIEW

File: Mlem/Views/Shared/SelectTextView.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                .background(Capsule().fill(Color.accentColor))
+                Button {
+                    dismiss()
+                } label: {
+                    Image(systemName: "xmark.circle.fill")
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(height: 30)
+                        .symbolRenderingMode(.palette)
+                        .foregroundStyle(.secondary, .secondary.opacity(0.2))
```
Comment (EricBAndrews) [inline Mlem/Views/Shared/SelectTextView.swift:44]: This introduces an inconsistency with our image viewer sheet, which feels similar enough that we should keep them aligned. The close/cancel/dismiss action is also almost always in the top left--it's a bit weird to see it over on the right.

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Looks good! Functionality all works.

Minor note on button placement/styling.

File: Mlem/Views/Shared/SelectTextView.swift (hunk 1 above)
Comment (Sjmarf) [reply Mlem/Views/Shared/SelectTextView.swift:44]: I put it in the top right for consistency with what iOS uses (share sheet, maps, etc). As far as I can tell they use this design everywhere Apple isn't entirely consistent with "cancel" buttons, though - they appear to be in the top-right when possible, or in the top-left if a "done" action is placed in the top-right.

If we want to remain consistent, maybe the image close button would be better on the right too?

Comment (EricBAndrews) [reply Mlem/Views/Shared/SelectTextView.swift:44]: Moving the image viewer close button works for me--I agree that the apple default seems to prefer the right unless it's taken by a send.

Comment (Sjmarf) [reply Mlem/Views/Shared/SelectTextView.swift]: Done 👍

**Review by EricBAndrews — APPROVED** [review]
Nice 👍

**Pull request merged**